### PR TITLE
fix(sshd): unlock the repository when gitreceive succeeds or fails

### PR DESCRIPTION
**Pull request by kmala**
# Summary of Changes

Currently the builder is not unlocking the repo if the git push succeeds or fails because of which we can't do multiple releases to any app

# Issue(s) that this PR Closes

# Associated [End To End Test](https://github.com/deis/workflow-e2e) PR(s)

# Associated [Documentation](https://github.com/deis/workflow) PR(s)

# Associated [Design Document](http://docs.deis.io/en/latest/contributing/design-documents)(s)

# Testing Instructions

Please provide a detailed list for how to test the changes in this PR.

1. Create a Deis Cluster
2. Register an app
3. do git push again or do a new release and you should succeed.

# Pull Request Hygiene TODOs

Please make sure the below checklist is complete.

- [x] Your pull request is concise and to the point (make another PR for refactoring nearby code)
- [x] Your commits follow the [commit style guidelines](http://docs.deis.io/en/latest/contributing/standards/#commit-style-guide)

### CODE REVIEW

File: pkg/sshd/lock_test.go [hunk 1]
```diff
@@ -12,6 +13,10 @@ const (
 	callbackTimeout = 1 * time.Second
 )
 
+var (
+	gitError = errors.New("git recieve error")
```
Comment (mboersma) [inline pkg/sshd/lock_test.go]: Typo: "receive". Also I notice the commit message has a few typos. Should be:
`fix(sshd): unlock the repository when gitreceive succeeds or fails`

File: pkg/sshd/lock.go [hunk 2]
```diff
@@ -38,9 +38,9 @@ func wrapInLock(lck RepositoryLock, repoName string, fn func() error) error {
 		case <-doneCh:
 		}
 	}()
+	defer lck.Unlock(repoName)
```
Comment (arschles) [inline pkg/sshd/lock.go]: @kmala this is definitely a good change, but I wasn't previously failing to `git push` a second time

Comment (kmala) [inline pkg/sshd/lock.go]: which commit of builder are using..because it used to work fine and was broken in https://github.com/deis/builder/pull/354

Comment (arschles) [inline pkg/sshd/lock.go]: ahh, good catch

**Comment by arschles** [comment]
LGTM3

**Pull request merged**